hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSNamesystem.java

```diff
@@ -2006,8 +2006,14 @@ private void metaSave(PrintWriter out) {
           continue;
         }
         Preconditions.checkState(ucFile instanceof INodeFile);
-        openFileIds.add(ucFileId);
+
         INodeFile inodeFile = ucFile.asFile();
```

Comment:
the set `dataNode.getLeavingServiceStatus().getOpenFiles()` is maintained by the decommissioning admin thread, so race condition is possible.

Reply:
so yeah, it's possible.